Unittests

matkrak opened the pull request:
Sorry for the mess with the repository. I was trying to work with git console and bad things happened.

Please check unittests.py file where I put some tests for Curve class.
I am writing tests for Profiles objects right now.
-  There is probably better way to compare domains instead of listing them.
- If I rebased this branch you probably wouldn't see this trash commits, am I right?

matkrak committed:
vol1 elo

matkrak committed:
cleaning1

matkrak committed:
cleaning2

matkrak committed:
cleaning3

matkrak committed:
Delete __init__.py

matkrak committed:
Delete curve.py

matkrak committed:
Delete profile.py

matkrak committed:
Create unittests.py

A file where I will put all beprof unittests

matkrak committed:
added metadata testing

grzanka commented:
I see commit with "vol1 elo" message and I am curious what was the reason to name it like that.

grzanka commented:
Be more elaborate with tests - cover more corner cases. Add profile methods tests. You can also split tests among more files. Check setup and teardown method. Test is exception machinery works properly.

matkrak commented:
About this commits - I thought it was something I was doing locally. I made a lot of different commits just to see the results but wasn't aware some of them can be seen by the others - sorry! 
Working on the tests right now.

matkrak committed:
Profiles tests added

Working on Exceptions right now, as mentioned in the comment merged 18-cleaning methods and subtracting curves 
create new test possibilities

grzanka commented on beprof/unittests.py:
```diff
@@ -0,0 +1,115 @@
+        lpr.mirror()
+        lpr.mirror()
+        self.assertEqual(list(lpr.x), list(orgx))
+        self.assertEqual(list(lpr.y), list(orgy))
```
Investigate following numpy functions: array_equal, array_equiv, allclose

grzanka commented on beprof/unittests.py:
```diff
@@ -0,0 +1,115 @@
+
+class TestCurveMethods(unittest.TestCase):
+
+    def test_metadata_1(self):
```
I think this 3 checks in test_metadata_1 should go to 3 different functions. If the test fail it would be easier to find out why.

grzanka commented on beprof/unittests.py:
```diff
@@ -0,0 +1,115 @@
+        self.assertEqual(list(cur.rebinned(step=1, fixp=0.5).x),
+                         [0.5, 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5, 9.5]
+                         )
+class TestProfileMethods(unittest.TestCase):
```
It is very similar to TestCurve methods and seems to violate DRY

matkrak committed:
Added new class, modified existing

New class is resposible for metadata testing.
Therefore all metadata tests has been removed from other classes.

Not using list() which I knew was a bad idea atb, using np.array_equal and np.allclose instead.

Still there will be more tests after merging logging package and others.

matkrak commented:
Merging subtracting curves PR would provide more functions for testing purpose.
I will also add more malicious tests then.

grzanka commented:
Leave a message when its done

grzanka commented:
Closing, as promised. In case of resuming activity, open new PR.
